mlflow/tracking/__init__.py

```diff
@@ -169,6 +169,19 @@ def create_experiment(experiment_name):
     return _get_store().create_experiment(experiment_name)
 
 
+def get_or_create_experiment(experiment_name):
+    """
+    Finds an existing experiment or creates it if it does not exist. Always returns the experiment ID.
+    """
+    if experiment_name is None or experiment_name == "":
+        raise Exception("Invalid experiment name '%s'" % experiment_name)
+    experiment = _get_store().get_experiment_by_name(experiment_name)
```

Comment:
As @Cs4r mentioned, `get_experiment_by_name` is not defined in `abstract_store.py`, and is therefore not implemented by the remote storage API. It may be reasonable to add this, or we might eventually add a filters API to `list_experiments` (similar to AWS Describe* APIs). In order to avoid locking that API in, perhaps we could start by implementing this in terms of `list_experiments`, and later make it more efficient by adding a new lower-level API?

Reply:
Sure, I'll refactor the code to rely on `list_experiments` rather than using a functionality that isn't implemented in all storage adapter implementations. 

Seems a reasonable approach to me.